developer/src/kmc-analyze/src/osk-character-use/index.ts

```diff
@@ -26,7 +26,9 @@ export class AnalyzeOskCharacterUse {
           this.addStrings(this.scanTouchLayout(file));
           break;
         case KeymanFileTypes.Source.Project:
-          await this.analyzeProject(file);
+          if(analyzeProjects) {
+            await this.analyzeProject(file);
+          }
```

Comment:
`else`: warn / error, rather than failing silently?

Reply:
This is is the behaviour we want here -- if we are analyzing a project already, we don't want to check the .kpj file again. It's not an error.

Reply:
I thought I saw something in the code comments somewhere about how a .kpj should not include a(nother?) .kpj in its file listings.  That's what I thought this block was addressing.

Reply:
The .kpj itself shouldn't include another .kpj (but you could hand-craft one that did), but it's also possible to specify a folder which could include a .kpj so then everything could be re-processed. In both cases, we want to skip reprocessing.

Reply:
The latter one, sure.  The former one is the one I was concerned about, since I thought I'd seen a comment in one of the other recent PRs indicating that as a point of concern.  You'd have written that comment, though... so if that's not ringing any bells, I guess I was mistaken?